src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,181 @@
 import io.github.oblarg.oblog.Loggable;
 import io.github.oblarg.oblog.annotations.Log;
+
 import java.util.List;
-import java.util.function.DoubleSupplier;
-import org.photonvision.EstimatedRobotPose;
+import java.util.function.Supplier;
+
 import org.sciborgs1155.lib.failure.Fallible;
-import org.sciborgs1155.lib.failure.FaultBuilder;
 import org.sciborgs1155.lib.failure.HardwareFault;
-import org.sciborgs1155.robot.Constants;
-import org.sciborgs1155.robot.Robot;
+import org.sciborgs1155.robot.Ports.DrivePorts;
 
-public class Drive extends SubsystemBase implements Fallible, Loggable, AutoCloseable {
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
 
-  @Log private final ModuleIO frontLeft;
-  @Log private final ModuleIO frontRight;
-  @Log private final ModuleIO rearLeft;
-  @Log private final ModuleIO rearRight;
 
-  private final List<ModuleIO> modules;
+public class Drive extends SubsystemBase implements Loggable, AutoCloseable{
+  //creates StandardDriveMotor
+  StandardDriveMotor driveMotor = new StandardDriveMotor();
+//new formatting, check out StandardDriveMotor for more; just added so it is less messy in this document
+  private final CANSparkMax FRmotor = driveMotor.create(DrivePorts.FR_DRIVE_PORT);
+  private final CANSparkMax FLmotor = driveMotor.create(DrivePorts.FL_DRIVE_PORT);
+  private final CANSparkMax MRmotor = driveMotor.create(DrivePorts.MR_DRIVE_PORT);
+  private final CANSparkMax MLmotor = driveMotor.create(DrivePorts.ML_DRIVE_PORT);
+  private final CANSparkMax BRmotor = driveMotor.create(DrivePorts.BL_DRIVE_PORT);
+  private final CANSparkMax BLmotor = driveMotor.create(DrivePorts.BL_DRIVE_PORT);
 
-  // this should be a generic IMU class, once WPILib implements it
-  @Log private final WPI_PigeonIMU imu = new WPI_PigeonIMU(PIGEON);
+  private final CANSparkMax[] rightSparks = {FRmotor, MRmotor, BRmotor};
+  private final CANSparkMax[] leftSparks = {FLmotor, MLmotor, BLmotor};
 
-  public final SwerveDriveKinematics kinematics = new SwerveDriveKinematics(MODULE_OFFSET);
+  private final MotorControllerGroup rightMotors = new MotorControllerGroup(rightSparks);
+  private final MotorControllerGroup leftMotors = new MotorControllerGroup(leftSparks);
 
-  // Odometry and pose estimation
-  private final SwerveDrivePoseEstimator odometry;
+  // Gyros
+  @Log private final WPI_PigeonIMU gyro = new WPI_PigeonIMU(DrivePorts.GYRO_PORT);
+  //Right and Left side encoders
+  @Log private final RelativeEncoder rightEncoder = FRmotor.getEncoder();
+  @Log private final RelativeEncoder leftEncoder = FLmotor.getEncoder();
 
-  @Log private final Field2d field2d = new Field2d();
-  private final FieldObject2d[] modules2d;
+  @Log private double heading;
+  
+  @Log private final PIDController RdrivePID = new PIDController(DriveConstants.kP, DriveConstants.kI, DriveConstants.kD);
```

Comment:
For ergonomics, you should statically import DriveConstants by adding
```java
import static .........DriveConstants.*; // replace ... with the actual import
```
That way, the contents of `DriveContents` are in the namespace of this file, so your code would look more like
```java
new PIDController(kP, kI, kD)
```